### DISCUSSION TITLE
mutation doesn't invalidates the cache [#2]

### DISCUSSION THREAD

**Discussion by bn-piar** [reduxjs/redux-toolkit#2463] [Q&A]
Hi guys. Can't make `createOrder` mutation to trigger invalidation of `getOrder` endpoint.

```javascript

const ordersApi = createApi({
    reducerPath: 'ordersApi',
    baseQuery: fetchBaseQuery({ baseUrl: ROOT_ENDPOINT }),
    tagTypes: ['Orders', 'Order'],
    endpoints: (builder) => ({
        getOrder: builder.query<TOrder, string>({
            query: (id) => `/orders/${id}`,
            providesTags: (result) =>
                result
                    ? [
                          { type: 'Order', id: result.id },
                          { type: 'Order', id: 'ORDER' },
                      ]
                    : [{ type: 'Order', id: 'ORDER' }],
            transformResponse: ({ _id, ...order }: TOrderResponse) => order,
        }),
        // some other queries
        ...
        createOrder: builder.mutation<TOrder, TCreateOrderDto>({
            query: (body) => {
                return {
                    url: '/orders/create',
                    method: 'POST',
                    body,
                };
            },
            invalidatesTags: (result) =>
                result
                    ? [
                          { type: 'Order', id: result.id },
                          { type: 'Order', id: 'ORDER' },
                      ]
                    : [
                          { type: 'Order', id: 'LIST' },
                      ],
        }),
        // some other mutations 
       ...
    }),
});

```
`createOrder` creates a new Order object or modifies existing (by either adding new product to `productsInOrder` array or increasing `amount` property of existing array element).  

The order object looks like this
```javascript
{
    "status": "NEW",
    "vendor": {
        "defaultVendor": false,
        "companyName": "Mahamba",
        "contactPerson": "Bahamab",
        "id": "62b95b8e5603c45a7950de22"
    },
    "productsInOrder": [
        {
            "quantityToOrder": 12,
            "price": 12,
            "orderLimit": 12,
            "product": "62b95b9c5603c45a7950de26",
            "amount": 4
        }
    ],
    "orderId": 51,
    "total": 48,
    "id": "62ba6d60c784c34497b72652"
}
```
but for some reason when a new order is created neither `getOrders` nor `getOrder` endpoints get refetched and I don't understand why.. 
`getOrder`'s `providesTags` id is exactly the same as `createOrder`s `invalidatesTags` id.

`editOrder` and `deleteOrder` work just fine...

Is nested objects that get mutated the problem? 

How do I make sure `getOrder` is invalidated when a new `order` is created?

**Comment by phryneas** [comment]
did you add the middleware to your store? If you look at the cache entry in the RTK Query tab in the REdux devtools, does it have the tags you expect?

**Reply by bn-piar** [reply]
@phryneas hey, thanks for your response. Yes the middleware is added to my store. All other mutations (edit and delete) work as expected.

`getOrder` query does have the tags I expect. 

Couldn't figure out where to find tags that mutation invalidates but I believe they are correct too, I console logged `result` parameter of `invalidatesTags` function and it matches `id` of `getOrder` query `id`
